src/manager/report_builder.rs

```diff
@@ -246,16 +246,30 @@ lazy_static! {
             "# Downloads the file specified by `uri` if `uri` is a gs uri to a subdirectory of carrot_downloads named with `key`\n",
             "def download_file_if_gs(uri, key):\n",
             "    if isinstance(uri, str) and uri.startswith('gs://'):\n",
+            "        # Get the filename from the uri and ensure we're not overwriting an existing file\n",
+            "        filename = ensure_unique_filename(uri, key)\n",
             "        # Attempt to download with gsutil\n",
-            "        download_status = subprocess.Popen(['gsutil', 'cp', uri, f'carrot_downloads/{key}'])\n",
+            "        download_status = subprocess.Popen(['gsutil', 'cp', uri, f'carrot_downloads/{key}/{filename}'])\n",
             "        # If it failed, print an error message and exit\n",
             "        if download_status.wait() != 0:\n",
-            "            sys.exit(f\"gsutil terminated with an non-zero exit code when attempting to download {uri}\")\n",
+            "            sys.exit(f'gsutil terminated with an non-zero exit code when attempting to download {uri}')\n",
             "        # Return the location of the file\n",
-            "        return f'carrot_downloads/{key}/{uri[uri.rfind(\"/\")+1:]}'\n",
+            "        return f'carrot_downloads/{key}/{filename}'\n",
             "    # Return None if it's not a gs uri\n",
             "    return None\n",
-            "    \n",
+            "# Extracts the filename from uri, and returns the filename, possibly prepended with a number if that is necessary\n",
```

Comment:
I don't know if this will make a difference, but I'm a little worried in the context of a Jupyter notebook that declaring this function below its use above might cause an error.

Reply:
This works for me when I run it locally and within the job that generates reports.  I believe it's fine because that function isn't actually called until `mkdir_and_download_files` is called at the end of that cell

Reply:
AH!  OK, that works!